.github/workflows/build_all.yml

```diff
@@ -373,104 +373,196 @@ jobs:
+      DOCKER_FILE: lte/gateway/docker/services/c/Dockerfile
+    outputs:
+      digest: ${{ steps.agw-build-docker.outputs.digest }}
+    steps:
+      - run: echo "DOCKER_REGISTRY=agw-test.artifactory.magmacore.org" >> $GITHUB_ENV
+        if: github.ref == 'refs/heads/master'
+
+      - run: echo "Publishing images to ${{ env.DOCKER_REGISTRY }}"
+
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+
+      - uses: ./.github/workflows/composite/docker-builder-agw
+        id: agw-build-docker
         with:
-          name: pr
-          path: pr/
-      - name: Create release Tag
-        if: github.event_name == 'push'
+          REGISTRY_USERNAME: ${{ secrets.JFROG_USERNAME }}
+          REGISTRY_PASSWORD: ${{ secrets.JFROG_PASSWORD }}
+          REGISTRY: ${{ env.DOCKER_REGISTRY }}
+          IMAGE: ${{ env.DOCKER_IMAGE }}
+          DOCKERFILE: ${{ env.DOCKER_FILE }}
+
+  agw-container-build-python:
+    if: github.repository_owner == 'magma'
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_REGISTRY: agw-ci.artifactory.magmacore.org
+      DOCKER_IMAGE: agw_gateway_python
+      DOCKER_FILE: lte/gateway/docker/services/python/Dockerfile
+    outputs:
+      digest: ${{ steps.agw-build-docker.outputs.digest }}
+    steps:
+      - run: echo "DOCKER_REGISTRY=agw-test.artifactory.magmacore.org" >> $GITHUB_ENV
+        if: github.ref == 'refs/heads/master'
+
+      - run: echo "Publishing images to ${{ env.DOCKER_REGISTRY }}"
+
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+
+      - uses: ./.github/workflows/composite/docker-builder-agw
+        id: agw-build-docker
+        with:
+          REGISTRY_USERNAME: ${{ secrets.JFROG_USERNAME }}
+          REGISTRY_PASSWORD: ${{ secrets.JFROG_PASSWORD }}
+          REGISTRY: ${{ env.DOCKER_REGISTRY }}
+          IMAGE: ${{ env.DOCKER_IMAGE }}
+          DOCKERFILE: ${{ env.DOCKER_FILE }}
+
+  agw-container-build-go:
+    if: github.repository_owner == 'magma'
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_REGISTRY: agw-ci.artifactory.magmacore.org
+      DOCKER_IMAGE: gateway_go
+      DOCKER_FILE: feg/gateway/docker/go/Dockerfile
+    outputs:
+      digest: ${{ steps.agw-build-docker.outputs.digest }}
+    steps:
+      - run: echo "DOCKER_REGISTRY=agw-test.artifactory.magmacore.org" >> $GITHUB_ENV
+        if: github.ref == 'refs/heads/master'
+
+      - run: echo "Publishing images to ${{ env.DOCKER_REGISTRY }}"
+
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+
+      - uses: ./.github/workflows/composite/docker-builder-agw
+        id: agw-build-docker
+        with:
+          REGISTRY_USERNAME: ${{ secrets.JFROG_USERNAME }}
+          REGISTRY_PASSWORD: ${{ secrets.JFROG_PASSWORD }}
+          REGISTRY: ${{ env.DOCKER_REGISTRY }}
+          IMAGE: ${{ env.DOCKER_IMAGE }}
+          DOCKERFILE: ${{ env.DOCKER_FILE }}
+
+  agw-container-build-is-successful:
```

Comment:
Since we already have a sucess job here: Would it not be much more concise to use a matrix job instead of the almost identical jobs above?

Reply:
Have a look at the `bazel_build_and_test` job in the `bazel.yml` workflow:
```
    strategy:
      fail-fast: false
      matrix:
        include:
          - bazel-config: ""
            bazel-target: "//..."
          - bazel-config: "--config=asan"
            bazel-target: "`bazel query 'kind(cc.*, //...)'`"
          - bazel-config: "--config=production"
            bazel-target: "`bazel query 'kind(cc.*, //...)'`"
```

Reply:
and https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations

Reply:
Very well, this is a great idea! However, how would you manage outputs of those matrix jobs?
I need a distinctive output of each one.
I.e. The behaviour from https://github.com/magma/magma/pull/14302/files#diff-71a71e993dac5ebcaa1ad57402f04ee226b14121a8052e4f994706670fd65a1aR516-R519 needs to be recreated. Any ideas on that too?

Reply:
https://stackoverflow.com/questions/70287603/github-actions-dynamic-outputs-for-job-with-strategy-matrix seems to be a start, but not nice. And within composite actions are recommended as a solution.

Reply:
Also related: https://github.com/orgs/community/discussions/17245

Reply:
I am inclined to keep it as is with composite actions for now. Maybe, we can have a look at multiple outputs from matrix jobs some time together? :crystal_ball: